### PR TITLE
remove remote/distributed classes

### PR DESCRIPTION
**Pull request by missedone**
moving the remote/distributed TestNG classes to separate project.
basically, this is prototype how we can remote distributed testng out.

see supplementary PR here: https://github.com/testng-team/testng-remote/pull/14

### CODE REVIEW

File: src/main/java/org/testng/TestNG.java [hunk 1]
```diff
@@ -1090,8 +1061,17 @@ else if (m_masterfileName == null) {
...
+  /**
+   * Run the test suites.
+   * <p>
+   * This method can be overridden by subclass. <br/>
+   * For example, DistributedTestNG to run in master/slave mode according to commandline args. 
+   * </p>
+   * @return
+   * @since 6.9.11 when moving distributed/remote classes out into separate project
+   */
+  protected List<ISuite> runSuites() {
```
Comment (missedone) [inline src/main/java/org/testng/TestNG.java]: introduce new override-able method `runSuites` to abstract how to run the suites, this is helpful for DistributedTestNG to override this method to run master/slave mode according to command line args, see also: https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/distributed/src/main/java/org/testng/distributed/DistributedTestNG.java#L54

**Comment by missedone** [comment]
ping @cbeust , @juherr

**Comment by cbeust** [comment]
That's a huge change. If you're telling me it's mostly files being moved around, I'll take your work and I'll merge it (pending @juherr's approval).

**Comment by missedone** [comment]
yes, most of them just being moved.
the 'distributed testng' function was moved out of TestNG as well, it's available in testng-remote DistributedTestNG

File: src/main/java/org/testng/TestNG.java [hunk 2]
```diff
@@ -1577,8 +1555,6 @@ public void configure(Map cmdLineArgs) {
     result.xmlPathInJar = (String) cmdLineArgs.get(CommandLineArgs.XML_PATH_IN_JAR);
     result.junit = (Boolean) cmdLineArgs.get(CommandLineArgs.JUNIT);
     result.mixed = (Boolean) cmdLineArgs.get(CommandLineArgs.MIXED);
-    result.master = (String) cmdLineArgs.get(CommandLineArgs.MASTER);
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: Some warnings/exceptions are missing here to say that the master/slave feature moved into another jar.

**Comment by juherr** [comment]
Everything seems good if we don't miss to transfert some files ;)

But merging this PR will have 1 big side effect: https://github.com/testng-team/testng-remote/issues/1 will have to be closed before the next TestNG release. Without it, we won't have workarround for the removed features.

File: src/main/java/org/testng/remote/RemoteArgs.java [hunk 3]
```diff
@@ -1,17 +0,0 @@
-package org.testng.remote;
-
-import com.beust.jcommander.Parameter;
-
-public class RemoteArgs {
```
Comment (juherr) [inline src/main/java/org/testng/remote/RemoteArgs.java]: I don't see it into https://github.com/testng-team/testng-remote/tree/rm_distributed_classes/distributed/distributed/src/main/java/org/testng/remote

File: src/main/java/org/testng/xml/ResultContentHandler.java [hunk 4]
```diff
@@ -1,152 +0,0 @@
...
-
-import java.util.List;
-
-/**
- * Parses testng-result.xml, create TestResultMessages and send them back to the listener
- * as we encounter them.
- *
- * @author Cedric Beust <cedric@beust.com>
- */
-public class ResultContentHandler extends DefaultHandler {
```
Comment (juherr) [inline src/main/java/org/testng/xml/ResultContentHandler.java]: I don't find it on https://github.com/testng-team/testng-remote/tree/rm_distributed_classes

File: src/main/java/org/testng/xml/ResultXMLParser.java [hunk 5]
```diff
@@ -1,103 +0,0 @@
...
-import java.io.InputStream;
-
-/**
- * Parses testng-result.xml.
- *
- * @see ResultContentHandler
- *
- * @author Cedric Beust <cedric@beust.com>
- */
-public class ResultXMLParser extends XMLParser<Object> {
```
Comment (juherr) [inline src/main/java/org/testng/xml/ResultXMLParser.java]: I don't find it on https://github.com/testng-team/testng-remote/tree/rm_distributed_classes

**Comment by juherr** [comment]
Check: Except 2 classes, I found all others.

**Comment by missedone** [comment]
thanks @juherr 

i just added the two missing classes here:
https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/remote/remote/src/main/java/org/testng/xml/ResultXMLParser.java
https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/remote/remote/src/main/java/org/testng/xml/ResultContentHandler.java

note, for compatibility purpose, i modified ResultXMLParser.java to call parse method with reflection.

i'm wondering is ResultXMLParser still in used? i don't see any reference to it, @cbeust

**Comment by cbeust** [comment]
Looks like we're ready to merge, @missedone can you do a rebase? If not I'll merge manually.

**Comment by missedone** [comment]
hi @cbeust 
PR rebased
thanks

**Pull request merged**